src/libraries/System.Runtime.InteropServices/tests/System.Runtime.InteropServices.UnitTests/System/Runtime/InteropServices/Marshal/GetDelegateForFunctionPointerTests.cs

```diff
@@ -154,9 +153,10 @@ public void GetDelegateForFunctionPointer_CantCast_ThrowsInvalidCastException()
         }
 
         public delegate void GenericDelegate<T>(T t);
-        public delegate void NonGenericDelegate(string t);
-        public delegate void OtherNonGenericDelegate(string t);
+        public delegate void NonGenericDelegate(int i);
+        public delegate void OtherNonGenericDelegate(int i);
 
-        private static void Method(string s) { }
+        [UnmanagedCallersOnly]
```

Comment:
NativeAOT uses `UnmanagedFunctionPointerAttribute` to make delegate stubs available to the AOT compiler when they can't figure it out directly from arguments to `GetFunctionPointerForDelegate` or `GetDelegateForFunctionPointer`